support/events/message.go

```diff
@@ -0,0 +1,56 @@
+package events
+
+import (
+	"context"
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MessageCollector interface {
+	ErrorMessages(resource crclient.Object) ([]string, error)
+}
+
+type messageCollector struct {
+	ctx    context.Context
+	client crclient.Client
+}
+
+func NewMessageCollector(ctx context.Context, client crclient.Client) MessageCollector {
+	return &messageCollector{
+		ctx:    ctx,
+		client: client,
+	}
+}
+
+func (c *messageCollector) ErrorMessages(resource crclient.Object) ([]string, error) {
```

Comment:
So we are ok with clearing the status if there was an error and we had events, but they got ttld?

Reply:
true, I'd forgotten about that. Let me fix

Reply:
that turns out to be a little bit more complicated than I thought :) 
In the case of infra for example, we aggregate messages from different resources and put that in the condition. How do we know which part of the message string to preserve?
I'd like to address it as a follow-up. When I observed events having gone away was almost a day after the issue occurred, which means that this PR as is will provide value.